tests/integration/targets/openssh_cert/tasks/main.yml

```diff
@@ -355,6 +355,85 @@
       that:
         - rc_serial_number_ignored is not changed
       msg: OpenSSH certificate generation with omitted serial number is idempotent.
+  - name: Generate always valid cert using agent without key in agent (should fail)
+    openssh_cert:
+      type: user
+      signing_key: '{{ output_dir }}/id_key'
+      public_key: '{{ output_dir }}/id_key.pub'
+      path: '{{ output_dir }}/id_cert_with_agent'
+      use_agent: yes
+      valid_from: always
+      valid_to: forever
+    register: rc_no_key_in_agent
+    ignore_errors: yes
+    environment:
+      SSH_AUTH_SOCK: "{{ openssh_agent_sock }}"
+    when: openssh_version is version("7.6",">=")
```

Comment:
Instead of adding `when:` in every step, how about using `block:` with `when:`?

Reply:
Sorry, I hadn't actually used or come across the block keyword before, didn't realize how useful it would have been. I reorganized the tests and put them in a separate block.